apps/ejabberd/src/mongoose_client_api.erl

```diff
@@ -55,39 +51,37 @@ options(Req, State) ->
 to_json(Req, User) ->
     {<<"{}">>, Req, User}.
 
-maybe_authorize(<<"OPTIONS">>, Req, State) ->
-    {true, Req, State};
-maybe_authorize(_, Req, State) ->
-    Auth = cowboy_req:parse_header(<<"authorization">>, Req),
-    case Auth of
-        {ok, undefined, _} ->
-            make_unauthorized_response(Req, State);
-        {ok, AuthDetails, Req2} ->
-            do_authorize(AuthDetails, Req2, State)
+%%--------------------------------------------------------------------
+%% Authorization
+%%--------------------------------------------------------------------
+
+% @doc cowboy callback
+is_authorized(Req, State) ->
+    HTTPMethod = mongoose_api_common:get_http_method(Req),
+    Creds = mongoose_api_common:get_creds(Req),
+    AuthMethod = mongoose_api_common:get_auth_method(Req),
+    case check_password(Creds) andalso
+         mongoose_api_common:is_known_auth_method(AuthMethod) orelse
+         is_noauth_http_method(HTTPMethod) of
+        true ->
+            {User, _} = Creds,
+            {true, Req, State#{user => User, jid => jid:from_binary(User)}};
```

Comment:
JID is parsed twice. Here and in `check_password`. It may be done already in `is_authorized` and the function would fail if JID is invalid (like it was done before the refactor).